packages/vm/tests/api/runBlock.spec.ts

```diff
@@ -346,3 +346,70 @@ tape('runBlock() -> API return values', async (t) => {
     t.end()
   })
 })
+
+tape('runBlock() -> tx types', async (t) => {
+  async function simpleRun(vm: VM, transactions: TypedTransaction[]) {
+    const common = vm._common
+
+    const blockRlp = testData.blocks[0].rlp
+    const block = Block.fromRLPSerializedBlock(blockRlp, { common, freeze: false })
+
+    //@ts-ignore overwrite transactions
+    block.transactions = transactions
+
+    //@ts-ignore
+    await setupPreConditions(vm.stateManager._trie, testData)
+
+    const res = await vm.runBlock({
+      block,
+      skipBlockValidation: true,
+      generate: true,
+    })
+
+    t.ok(
+      res.gasUsed.eq(
+        res.receipts
+          .map((r) => r.gasUsed)
+          .reduce((prevValue: BN, currValue: Buffer) => prevValue.add(new BN(currValue)), new BN(0))
+      ),
+      "gas used should equal transaction's total gasUsed"
+    )
+  }
+
+  t.test('legacy tx', async (t) => {
+    const common = new Common({ chain: 'mainnet', hardfork: 'berlin' })
+    const vm = setupVM({ common })
+    await vm.stateManager.generateCanonicalGenesis()
+
+    const tx = Transaction.fromTxData({ gasLimit: 53000, value: 1 }, { common, freeze: false })
+
+    // set `from` to a genesis address with existing balance
+    const address = Address.fromString('0xccfd725760a68823ff1e062f4cc97e1360e8d997')
+    tx.getSenderAddress = () => {
+      return address
+    }
+
+    await simpleRun(vm, [tx])
+    t.end()
+  })
+
+  t.test('access list tx', async (t) => {
+    const common = new Common({ chain: 'mainnet', hardfork: 'berlin' })
+    const vm = setupVM({ common })
+    await vm.stateManager.generateCanonicalGenesis()
+
+    const tx = AccessListEIP2930Transaction.fromTxData(
+      { gasLimit: 53000, value: 1, v: 1, r: 1, s: 1 },
+      { common, freeze: false }
+    )
+
+    // set `from` to a genesis address with existing balance
+    const address = Address.fromString('0xccfd725760a68823ff1e062f4cc97e1360e8d997')
+    tx.getSenderAddress = () => {
+      return address
+    }
+
+    await simpleRun(vm, [tx])
+    t.end()
+  })
+})
```

Comment:
So, for the release notes: did `runBlock()` previously not work with typed txs or are these just additional tests for safety reasons?

Reply:
there was a bug in the block constructor would recreate a AccessListEIP2930Transaction as a legacy transaction (due to the missing `type` field, solved by the newly introduced alias). these tests were added to ensure the bug was fixed, and so it won't be reintroduced in the future (https://github.com/ethereumjs/ethereumjs-monorepo/pull/1185#issuecomment-813735865)